egui/src/containers/scroll_area.rs

```diff
@@ -239,17 +242,15 @@ impl Prepared {
 
             if response.active {
                 if let Some(mouse_pos) = ui.input().mouse.pos {
-                    if handle_rect.contains(mouse_pos) {
-                        if inner_rect.top() <= mouse_pos.y && mouse_pos.y <= inner_rect.bottom() {
-                            state.offset.y +=
-                                ui.input().mouse.delta.y * content_size.y / inner_rect.height();
-                        }
-                    } else {
-                        // Center scroll at mouse pos:
-                        let mpos_top = mouse_pos.y - handle_rect.height() / 2.0;
-                        state.offset.y = remap(mpos_top, top..=bottom, 0.0..=content_size.y);
-                    }
+                    let scroll_start_offset_from_top = state
+                        .scroll_start_offset_from_top
+                        .get_or_insert_with(|| mouse_pos.y - handle_rect.top());
```

Comment:
```suggestion
                    let scroll_start_offset_from_top = state
                        .scroll_start_offset_from_top
                        .get_or_insert_with(|| {
                            if handle_rect.contains(mouse_pos) {
                                mouse_pos.y - handle_rect.top()
                            } else {
                                0.5 * handle_rect.height() // center handle on mouse
                            }
                        });
```

Reply:
Good catch with an easy fix :100:

Reply:
Made a more complete fix, that deal with the cases when the handle centering does not occur, ex: clicking near the inner_rect bottom or top.